### PR TITLE
Clean up warnings and bump version to 0.1.9 for development

### PR DESCRIPTION
**Pull request by nomadbyte**
This merges updates and history from the [nomadbyte/develop][develop] branch to continue with this line of changes to the baseline STOICAL-0.1.8.

Briefly:

- The version number is bumped to 0.1.9 to keep all recent development changes separate from the 0.1.8 release.
- The project now builds with only a single warning (at `-Wall`), which is for a set but unused variable in RegEx handling.
  Apparently this part was still under development in 0.1.8. So it makes sense to keep it for now as a reminder to revisit that
  code at some point.
- The examples/threads now seems to work properly (to my understanding) -- a "column of fish" swims in the set region of the
   terminal (LINES/COLUMNS). Randomly some fishes are somewhat leading the others. Can't tell how it's supposed to be, but
   a few updates in thread delay handling were needed to get it to work (in xterm too).

Generally, the code seems to have aged not too badly, well, it seems to work to the designed specs. However, at this point it **needs extensive exercising**, so that there could be established some understanding of what it could and should do.

**Are you familiar with STOICAL or STOIC dialect enough to cook up some scripts that would demo some basic usage?**

 Perhaps something even beyond the arithmetic and more with its advanced constructs like:
- use of dictionaries
- code
- iterations
- use of type-system
- files and sockets etc. 

Basically, some practical examples for all constructs mentioned in the Stoical manual.

I'm not versed in this dialect, and reading the included Stoical manual helped me just a little. So having some ready code could help us iron out some remaining issues.

[develop]: https://github.com/nomadbyte/stoical-fixes/tree/develop

### CODE REVIEW

**Review by mikewarot — APPROVED** [review]
wow... thanks for all your work

**Comment by nomadbyte** [comment]
I wonder if you had a chance to actually build and run the updated code?

**Comment by mikewarot** [comment]
I built and ran it under windows 10/windows subsystem for Linux. It worked!

I'm digging up all the documentation and old examples I can find. I am not
adverse to changing anything that seems irrational about the language, if
you happen upon something like that.

I thought about the idea of making it more FIG-forth like...but if I do
that people will expect compatibility, which isn't an option.

**Comment by nomadbyte** [comment]
I agree it does not make sense to change the language too much at this stage.

Some tweaks will definitely be needed to have some practical consistency.  I'm currently writing some tests just to make sense of the language and also fix some expectations. Already I bumped into some "imperfections" of the defined behavior. Among others is the behavior of "feq" operator, which is not really floating-point aware, just doing the exact not epsilon-based compare.

Also, looks like the author intended to create some STOIC compatibility definitions (empty now). It may be possible to accomplish that too, now that you located the STOIC doc. Nice find!
